File: ruby/j20.arguments.rb
#!/usr/bin/ruby
arg_count = ARGV.length   # get num of arguments
first     = 0             # set index of first element
last      = arg_count - 1 # set index of last element

puts "The arguments passed are (reverse order):"
for count in last.downto(first)           # decrement from # of args to 0 
  puts " item #{count+1}: #{ARGV[count]}" # count & correponding ARGV element
end  

File: python/j20.arguments.py
#!/usr/bin/python
import sys

# illustrative variables 
ARG_COUNT = len(sys.argv) - 1 # get num of real arguments
FIRST     = 1                 # index of first argument
# utility variables
count     = ARG_COUNT         # initialize counter

print "The arugments passed are:"
# use count style loop
while count >= FIRST:
   arg = sys.argv[count]  # get arg using index
   # print count and argument using count index
   print " item %d: %s" % (count,arg) 
   count -= 1             # decrement counter

